vscode/microsoft-kiota/src/kiotaInterop/install.ts

```diff
@@ -0,0 +1,151 @@
+import AdmZip from 'adm-zip';
+import appdataPath from 'appdata-path';
+import { https } from "follow-redirects";
+import * as fs from 'fs';
+import * as path from 'path';
+
+import runtimeJson from './runtime.json';
+
+const kiotaInstallStatusKey = "kiotaInstallStatus";
+const installDelayInMs = 30000; // 30 seconds
+const state: { [key: string]: any } = {};
+
+let kiotaPath: string | undefined;
+const binariesRootDirectory = '.kiotabin';
+const baseDownloadUrl = "https://github.com/microsoft/kiota/releases/download";
+
+export interface Package {
+  platformId: string;
+  sha256: string;
+}
+
+const windowsPlatform = 'win';
+const osxPlatform = 'osx';
+const linuxPlatform = 'linux';
+
+const baseDir = appdataPath('Microsoft Kiota');
+
+async function runIfNotLocked(action: () => Promise<void>) {
+  const installStartTimeStamp = state[kiotaInstallStatusKey];
+  const currentTimeStamp = new Date().getTime();
+  if (!installStartTimeStamp || (currentTimeStamp - installStartTimeStamp) > installDelayInMs) {
+    //locking the context to prevent multiple downloads across multiple instances
+    //overriding after 30 seconds to prevent stale locks
+    state[kiotaInstallStatusKey] = currentTimeStamp;
+    try {
+      await action();
+    } finally {
+      state[kiotaInstallStatusKey] = undefined;
+    }
+  }
+}
+
+export async function ensureKiotaIsPresent() {
+  const runtimeDependencies = getRuntimeDependenciesPackages();
+  const currentPlatform = getCurrentPlatform();
+  const installPath = getKiotaPathInternal(false);
+  if (installPath) {
+    if (!fs.existsSync(installPath) || fs.readdirSync(installPath).length === 0) {
```

Comment:
Consider using the async APIs instead of the sync APIs. I think the right APIs here should be `fs.stat` and `fs.readdir` respectively

Reply:
What may have been the motivation of doing them sync in the first place? 
I intended to have the code changes to mainly move functions from one location to another without changing the overall functionality so that I do not introduce any unintended consequences

Reply:
I'm actually not sure. You can create an issue for this and work on it in a later PR

Reply:
I've created the issue here for tracking: https://github.com/microsoft/kiota/issues/6165